backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1487,3 +1487,77 @@ databaseChangeLog:
                     nullable: false
                     foreignKeyName: fk__patient_link__time_of_consent
                     references: patient_link
+  - changeSet:
+      id: add-audit-event-table
+      author: bwarfield@cdc.gov
+      comment: Add table for API request auditing.
+      changes:
+        - createTable:
+            tableName: api_audit_event
+            remarks: An API access event that we record for HIPAA auditing.
+            columns:
+              - column:
+                  name: id
+                  type: *idtype
+                  remarks: The internal database identifier for this audit event.
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: event_timestamp
+                  type: DATETIME
+                  remarks: The creation timestamp for this entity.
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_id
+                  remarks: The correlation ID from the application logs.
+                  type: *string
+              - column:
+                  name: http_request_details
+                  remarks: Key details of the HTTP request.
+                  type: jsonb
+                  constraints:
+                    nullable: false
+              - column:
+                  name: graphql_query_details
+                  remarks: Lightly structured information about the graphql request.
+                  type: jsonb
+              - column:
+                  name: graphql_error_paths
+                  remarks: The list of paths in the graphql response that had errors (usually empty).
+                  type: "text[]"
+              - column:
+                  name: api_user_id
+                  type: *idtype
+                  remarks: The user who performed this action.
+                  constraints:
+                    nullable: false # TODO: decide if patient users should be autocreated for this
```

Comment:
`# TODO:` spotted. I'm ok if it stays, just calling it out just in case.

Reply:
It's an open question: technically speaking, we don't need an ApiUser object for a patient until they update an audited record (Person or AoeSurvey). But that seems silly, since any patient who makes any request is almost certain to eventually do one of those things, so I've mostly talked myself into requiring this to be present at all times. (Change my mind!)